### PR TITLE
Paypal payment

### PR DESCRIPTION
**Pull request by GustavoDinizMonteiro**
Add a generic component to provide Paypal payment for platform operations

### CODE REVIEW

**Comment by GustavoDinizMonteiro** [comment]
Depends of #44

File: src/app/essay/create-essay/create-essay.component.ts [hunk 1]
```diff
@@ -45,8 +53,13 @@ export class CreateEssayComponent implements OnInit {
                 }
             );
 
-        const user: User = JSON.parse(sessionStorage.getItem('currentUser'));
-        if (user.usingWeekelyTopic) {
+        this.user = JSON.parse(sessionStorage.getItem('currentUser'));
+
+        if(this.user.role === 'Free') {
+            this.paymentValue = this.singleEssayPrice;
+        }
```
Comment (GustavoDinizMonteiro) [inline src/app/essay/create-essay/create-essay.component.ts]: will we have both types of premium users @Benardi ?

File: src/app/essay/create-essay/create-essay.component.ts [hunk 2]
```diff
@@ -73,7 +86,20 @@ export class CreateEssayComponent implements OnInit {
             .subscribe(
                 (essay: Essay) => {
                     this.essayService.addEssayElement(essay);
-                    this.onEndSubmission();
+
+                    if(paid) {
+                        let record = {
+                            essayId: essay.id,
+                            value: this.paymentValue * 0.2  // TODO: Check this value.
```
Comment (GustavoDinizMonteiro) [inline src/app/essay/create-essay/create-essay.component.ts]: could you check and pass me the percentages/amounts paid to the editors for essays for each type of user @Benardi ?

**Comment by GustavoDinizMonteiro** [comment]
Depends of https://github.com/corrigeai/corrigeai-api/pull/42

**Comment by GustavoDinizMonteiro** [comment]
You can review now @Benardi :)

**Comment by Benardi** [comment]
Assuming the problem in the submission of a new essay doesn't concern this PR (as it's another feature)  everything looks good. The paypal is also working in the mobile now.

**Pull request merged**